annif/rest.py

```diff
@@ -248,6 +258,11 @@ def learn(
     try:
         corpus = _documents_to_corpus(body, project.subjects)
         project.learn(corpus)
+    except OperationFailedException as err:
+        if str(err) == "Learning not enabled for project":
```

Comment:
If a custom exception was used to indicate disabled functionality, instead of OperationFailedException, there would be no need to perform the fragile string comparison here.